packages/babel-plugin-transform-regenerator/src/regenerator/emit.ts

```diff
@@ -56,1288 +30,1301 @@ const Ep = Emitter.prototype;
 
-          self.emit(
-            t.returnStatement(
-              t.callExpression(self.contextProperty("finish"), [
-                finallyEntry.firstLoc,
-              ]),
-            ),
-          );
-        }
-      });
+          if (finallyLoc) {
+            self.updateContextPrevLoc(self.mark(finallyLoc));
 
-      self.mark(after);
+            self.leapManager.withEntry(finallyEntry, function () {
+              self.explodeStatement(path.get("finalizer"));
+            });
 
-      break;
+            self.emit(
+              t.returnStatement(
+                t.callExpression(self.contextProperty("finish"), [
+                  finallyEntry.firstLoc,
+                ]),
+              ),
+            );
+          }
+        });
 
-    case "ThrowStatement":
-      self.emit(t.throwStatement(self.explodeExpression(path.get("argument"))));
+        self.mark(after);
 
-      break;
+        break;
 
-    case "ClassDeclaration":
-      self.emit(self.explodeClass(path));
-      break;
+      case "ThrowStatement":
+        self.emit(
+          t.throwStatement(self.explodeExpression(path.get("argument"))),
+        );
 
-    default:
-      throw new Error("unknown Statement of type " + JSON.stringify(stmt.type));
-  }
-};
+        break;
 
-const catchParamVisitor = {
-  Identifier: function (path: any, state: any) {
-    if (path.node.name === state.catchParamName && util.isReference(path)) {
-      util.replaceWithOrRemove(path, state.getSafeParam());
-    }
-  },
+      case "ClassDeclaration":
+        self.emit(self.explodeClass(path));
+        break;
 
-  Scope: function (path: any, state: any) {
-    if (path.scope.hasOwnBinding(state.catchParamName)) {
-      // Don't descend into nested scopes that shadow the catch
-      // parameter with their own declarations.
-      path.skip();
+      default:
+        throw new Error(
+          "unknown Statement of type " + JSON.stringify(stmt.type),
+        );
     }
-  },
-};
-
-Ep.emitAbruptCompletion = function (record: any) {
-  if (!isValidCompletion(record)) {
-    assert.ok(false, "invalid completion record: " + JSON.stringify(record));
   }
 
-  assert.notStrictEqual(
-    record.type,
-    "normal",
-    "normal completions are not abrupt",
-  );
+  emitAbruptCompletion(record: Completion) {
+    assert.notStrictEqual(
+      record.type,
+      "normal",
+      "normal completions are not abrupt",
+    );
 
-  const t = util.getTypes();
-  const abruptArgs = [t.stringLiteral(record.type)];
-
-  if (record.type === "break" || record.type === "continue") {
-    t.assertLiteral(record.target);
-    abruptArgs[1] = this.insertedLocs.has(record.target)
-      ? record.target
-      : t.cloneDeep(record.target);
-  } else if (record.type === "return" || record.type === "throw") {
-    if (record.value) {
-      t.assertExpression(record.value);
-      abruptArgs[1] = this.insertedLocs.has(record.value)
-        ? record.value
-        : t.cloneDeep(record.value);
```

Comment:
Replace it with `abruptArgs[1] = t.cloneNode(record.value);`, because I think there is only `t.NumericLiteral` in `insertedLocs`.